tedge/src/cli/connect/command.rs

```diff
@@ -88,21 +94,23 @@ impl Command for ConnectCommand {
             &self.config_location,
         )?;
 
-        if self.check_connection(&config).is_err() {
+        let connection_check_result = self.check_connection(&config);
+        if connection_check_result.is_err() {
             println!(
                 "Warning: Bridge has been configured, but {} connection check failed.\n",
                 self.cloud.as_str()
             );
         }
 
         if let Cloud::C8y = self.cloud {
-            println!("Restarting mosquitto to resubscribe to bridged inbound cloud topics after device creation");
-            restart_mosquitto(
-                &bridge_config,
-                self.service_manager.as_ref(),
-                &self.config_location,
-            )?;
-
+            if let Ok(DeviceStatus::New) = connection_check_result {
```

Comment:
The check MUST be replaced by  status != DeviceStatus::AlreadyExists.

We have no way to be sure that the device has been already registered.